linera-execution/src/evm/revm.rs

```diff
@@ -287,6 +302,50 @@ enum PrecompileTag {
     MessageIsBouncing,
 }
 
+/// Some functionalities from the ServiceRuntime not in BaseRuntime
+#[derive(Debug)]
+enum ServicePrecompileTag {
+    /// Try query application
+    TryQueryApplication,
+}
+
+/// Key prefixes used to transmit precompiles.
+#[derive(Debug)]
+enum PrecompileTag {
+    Base(BasePrecompileTag),
+    Contract(ContractPrecompileTag),
+    Service(ServicePrecompileTag),
+}
+
+impl PrecompileTag {
+    fn from_u8(tag: u8) -> Result<Self, String> {
```

Comment:
Is there a way we could avoid this boilerplate?

Reply:
(For instance using `bincode` or `bcs` serialization)

Reply:
Yes, there are ways, but they are not necessarily as simple as 10 lines.

Reply:
We consistently avoid boilerplate in the rest of the codebase

Reply:
Ok, changed.